core/src/test/java/org/elasticsearch/cluster/MinimumMasterNodesIT.java

```diff
@@ -273,12 +273,14 @@ public void testDynamicUpdateMinimumMasterNodes() throws Exception {
                 .put("discovery.initial_state_timeout", "500ms")
                 .build();
 
-        logger.info("--> start 2 nodes");
-        internalCluster().startNodesAsync(2, settings).get();
+        logger.info("--> start first node and wait for it to be a master");
+        internalCluster().startNode(settings);
+        ensureClusterSizeConsistency();
 
         // wait until second node join the cluster
-        ClusterHealthResponse clusterHealthResponse = client().admin().cluster().prepareHealth().setWaitForEvents(Priority.LANGUID).setWaitForNodes("2").get();
-        assertThat(clusterHealthResponse.isTimedOut(), equalTo(false));
+        logger.info("--> start second node and wait for it to join");
+        internalCluster().startNode(settings);
+        ensureClusterSizeConsistency();
```

Comment:
These changes are needed because the test start two nodes concurrently without setting min master nodes. This can cause split brain - this can happen normally too but switching to MockZenPings (to speed up tests) makes it much more likely. Instead of running the risk I changed the test to start nodes one at a time.